feature-libs/order/components/order-details/order-detail-reorder/order-detail-reorder.component.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Component,
+  ElementRef,
+  OnDestroy,
+  OnInit,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import { ActiveCartService } from '@spartacus/cart/base/core';
+import { CartModificationList, CartOutlets } from '@spartacus/cart/base/root';
+import { UserIdService } from '@spartacus/core';
+import { ReorderOrderFacade } from '@spartacus/order/root';
+import { LaunchDialogService, LAUNCH_CALLER } from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { OrderDetailsService } from '../order-details.service';
+
+@Component({
+  selector: 'cx-order-details-reorder',
+  templateUrl: './order-detail-reorder.component.html',
+})
+export class OrderDetailReorderComponent implements OnInit, OnDestroy {
+  constructor(
+    protected orderDetailsService: OrderDetailsService,
+    protected launchDialogService: LaunchDialogService,
+    protected vcr: ViewContainerRef,
+    protected userIdService: UserIdService,
+    protected reorderOrderFacade: ReorderOrderFacade,
+    protected activeCartService: ActiveCartService
+  ) {}
+
+  @ViewChild('element') element: ElementRef;
+  protected subscription = new Subscription();
+  order$: Observable<any>;
+  userId: string;
+  readonly CartOutlets = CartOutlets;
+
+  ngOnInit() {
+    this.order$ = this.orderDetailsService.getOrderDetails();
+    this.subscription.add(
+      this.userIdService.getUserId().subscribe((userId) => {
+        this.userId = userId;
+      })
+    );
+  }
+
+  onReorderClick(order: any) {
+    this.launchDialog();
+    this.reorderOrderFacade
```

Comment:
Instead, of launching the dialog and then manage the reorder process here, It seems like it would simplify the implementation to simply open the dialog and let it launch and manage the reorder process.  What do you think?